packages/spatial/src/camera/systems/FollowCameraInputSystem.ts

```diff
@@ -83,19 +99,206 @@ const onFollowCameraShoulderCam = (cameraEntity: Entity) => {
 }
 
 /**
- * Change camera distance.
+ * Change camera distance or navigate between POIs.
  * @param cameraEntity Entity holding camera and input component.
+ * @param axes Input axes values.
+ * @param deltaTime Delta time for smooth transitions.
  */
 export const handleFollowCameraScroll = (
   cameraEntity: Entity,
   axes: AxisValueMap<typeof DefaultAxisBindings>,
   deltaTime: number
 ): void => {
   const follow = getComponent(cameraEntity, FollowCameraComponent)
-
   const zoomDelta = axes.FollowCameraZoomScroll ?? 0
   const shoulderDelta = axes.FollowCameraShoulderCamScroll ?? 0
 
+  // Check if we're in POI mode
+  const cameraSettingsState = getMutableState(CameraSettingsState)
+
+  if (cameraSettingsState.cameraMode.value === CameraMode.POI) {
```

Comment:
rather than overloading our existing follow camera component with non-follow functionality, lets move this to it's own component & system. this is what composition is all about!